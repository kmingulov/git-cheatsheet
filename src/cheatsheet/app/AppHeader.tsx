import React, { Component, ReactElement } from 'react';
import { Container, Nav, Navbar } from 'react-bootstrap';
import { GoMarkGithub } from 'react-icons/go';

/**
 * Header component with application's name.
 * @inheritdoc
 */
export class AppHeader extends Component {
  public shouldComponentUpdate(): boolean {
    return false;
  }

  public render(): ReactElement {
    return (
      <Navbar bg='primary' sticky='top'>
        <Container className='header'>
          <Navbar.Brand>Git Cheatsheet</Navbar.Brand>

          <Navbar.Collapse>
            <Nav className='mr-auto'>
              <Nav.Link>
                About
              </Nav.Link>
            </Nav>
            <Nav>
              <Nav.Link href='https://github.com/kmingulov/git-cheatsheet' target='_blank'>
                <GoMarkGithub size={ 24 }/>
              </Nav.Link>
            </Nav>
          </Navbar.Collapse>
        </Container>
      </Navbar>
    );
  }
}
