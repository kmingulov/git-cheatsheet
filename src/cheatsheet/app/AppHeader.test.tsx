import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';

import { AppHeader } from './AppHeader';

Enzyme.configure({adapter: new Adapter()});

describe(AppHeader, () => {
  it('renders correctly', () => {
    const component = mount(<AppHeader/>);
    expect(component).toMatchSnapshot();

    component.setProps({ prop: 'value' });
    expect(component).toMatchSnapshot();
  });
});
