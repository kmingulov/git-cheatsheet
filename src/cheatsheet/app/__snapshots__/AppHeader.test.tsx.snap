// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader renders correctly 1`] = `
<AppHeader>
  <ForwardRef(Bootstrap(undefined))
    bg="primary"
    sticky="top"
  >
    <ForwardRef
      bg="primary"
      bsPrefix="navbar"
      innerRef={null}
      sticky="top"
    >
      <Uncontrolled(Navbar)
        bg="primary"
        bsPrefix="navbar"
        innerRef={null}
        sticky="top"
      >
        <Navbar
          as="nav"
          bg="primary"
          bsPrefix="navbar"
          collapseOnSelect={false}
          expand={true}
          onToggle={[Function]}
          sticky="top"
          variant="light"
        >
          <nav
            className="navbar navbar-expand navbar-light bg-primary sticky-top"
          >
            <ForwardRef(Bootstrap(Container))
              className="header"
            >
              <Container
                as="div"
                bsPrefix="container"
                className="header"
                fluid={false}
              >
                <div
                  className="header container"
                >
                  <ForwardRef(Bootstrap(NavbarBrand))>
                    <NavbarBrand
                      bsPrefix="navbar-brand"
                    >
                      <span
                        className="navbar-brand"
                      >
                        Git Cheatsheet
                      </span>
                    </NavbarBrand>
                  </ForwardRef(Bootstrap(NavbarBrand))>
                  <ForwardRef(Bootstrap(NavbarCollapse))>
                    <NavbarCollapse
                      bsPrefix="navbar-collapse"
                    >
                      <Collapse
                        appear={false}
                        dimension="height"
                        getDimensionValue={[Function]}
                        in={false}
                        mountOnEnter={false}
                        timeout={300}
                        unmountOnExit={false}
                      >
                        <Transition
                          addEndListener={[Function]}
                          appear={false}
                          aria-expanded={null}
                          enter={true}
                          exit={true}
                          in={false}
                          mountOnEnter={false}
                          onEnter={[Function]}
                          onEntered={[Function]}
                          onEntering={[Function]}
                          onExit={[Function]}
                          onExited={[Function]}
                          onExiting={[Function]}
                          timeout={300}
                          unmountOnExit={false}
                        >
                          <div
                            aria-expanded={null}
                            className="navbar-collapse collapse"
                          >
                            <ForwardRef(ContextTransform)
                              className="mr-auto"
                            >
                              <ForwardRef
                                className="mr-auto"
                                navbar={true}
                                navbarBsPrefix="navbar"
                              >
                                <Uncontrolled(Component)
                                  className="mr-auto"
                                  innerRef={null}
                                  navbar={true}
                                  navbarBsPrefix="navbar"
                                >
                                  <ForwardRef(Bootstrap(Nav))
                                    className="mr-auto"
                                    navbar={true}
                                    navbarBsPrefix="navbar"
                                    onSelect={[Function]}
                                  >
                                    <Nav
                                      as="div"
                                      bsPrefix="nav"
                                      className="mr-auto"
                                      fill={false}
                                      justify={false}
                                      navbar={true}
                                      navbarBsPrefix="navbar"
                                      onSelect={[Function]}
                                    >
                                      <ForwardRef(ContextTransform(AbstractNav))
                                        as="div"
                                        className="mr-auto navbar-nav"
                                        onSelect={[Function]}
                                      >
                                        <AbstractNav
                                          as="div"
                                          className="mr-auto navbar-nav"
                                          onSelect={[Function]}
                                          parentOnSelect={[Function]}
                                        >
                                          <div
                                            className="mr-auto navbar-nav"
                                            onKeyDown={[Function]}
                                          >
                                            <ForwardRef(Bootstrap(NavLink))>
                                              <NavLink
                                                as={[Function]}
                                                bsPrefix="nav-link"
                                                disabled={false}
                                                innerRef={null}
                                              >
                                                <AbstractNavItem
                                                  as={[Function]}
                                                  className="nav-link"
                                                  disabled={false}
                                                >
                                                  <SafeAnchor
                                                    as="a"
                                                    className="nav-link"
                                                    data-rb-event-key={null}
                                                    disabled={false}
                                                    onClick={[Function]}
                                                  >
                                                    <a
                                                      className="nav-link"
                                                      data-rb-event-key={null}
                                                      href="#"
                                                      onClick={[Function]}
                                                      onKeyDown={[Function]}
                                                      role="button"
                                                    >
                                                      About
                                                    </a>
                                                  </SafeAnchor>
                                                </AbstractNavItem>
                                              </NavLink>
                                            </ForwardRef(Bootstrap(NavLink))>
                                          </div>
                                        </AbstractNav>
                                      </ForwardRef(ContextTransform(AbstractNav))>
                                    </Nav>
                                  </ForwardRef(Bootstrap(Nav))>
                                </Uncontrolled(Component)>
                              </ForwardRef>
                            </ForwardRef(ContextTransform)>
                            <ForwardRef(ContextTransform)>
                              <ForwardRef
                                navbar={true}
                                navbarBsPrefix="navbar"
                              >
                                <Uncontrolled(Component)
                                  innerRef={null}
                                  navbar={true}
                                  navbarBsPrefix="navbar"
                                >
                                  <ForwardRef(Bootstrap(Nav))
                                    navbar={true}
                                    navbarBsPrefix="navbar"
                                    onSelect={[Function]}
                                  >
                                    <Nav
                                      as="div"
                                      bsPrefix="nav"
                                      fill={false}
                                      justify={false}
                                      navbar={true}
                                      navbarBsPrefix="navbar"
                                      onSelect={[Function]}
                                    >
                                      <ForwardRef(ContextTransform(AbstractNav))
                                        as="div"
                                        className="navbar-nav"
                                        onSelect={[Function]}
                                      >
                                        <AbstractNav
                                          as="div"
                                          className="navbar-nav"
                                          onSelect={[Function]}
                                          parentOnSelect={[Function]}
                                        >
                                          <div
                                            className="navbar-nav"
                                            onKeyDown={[Function]}
                                          >
                                            <ForwardRef(Bootstrap(NavLink))
                                              href="https://github.com/kmingulov/git-cheatsheet"
                                              target="_blank"
                                            >
                                              <NavLink
                                                as={[Function]}
                                                bsPrefix="nav-link"
                                                disabled={false}
                                                href="https://github.com/kmingulov/git-cheatsheet"
                                                innerRef={null}
                                                target="_blank"
                                              >
                                                <AbstractNavItem
                                                  as={[Function]}
                                                  className="nav-link"
                                                  disabled={false}
                                                  href="https://github.com/kmingulov/git-cheatsheet"
                                                  target="_blank"
                                                >
                                                  <SafeAnchor
                                                    as="a"
                                                    className="nav-link"
                                                    data-rb-event-key="https://github.com/kmingulov/git-cheatsheet"
                                                    disabled={false}
                                                    href="https://github.com/kmingulov/git-cheatsheet"
                                                    onClick={[Function]}
                                                    target="_blank"
                                                  >
                                                    <a
                                                      className="nav-link"
                                                      data-rb-event-key="https://github.com/kmingulov/git-cheatsheet"
                                                      href="https://github.com/kmingulov/git-cheatsheet"
                                                      onClick={[Function]}
                                                      onKeyDown={[Function]}
                                                      target="_blank"
                                                    >
                                                      <GoMarkGithub
                                                        size={24}
                                                      >
                                                        <div />
                                                      </GoMarkGithub>
                                                    </a>
                                                  </SafeAnchor>
                                                </AbstractNavItem>
                                              </NavLink>
                                            </ForwardRef(Bootstrap(NavLink))>
                                          </div>
                                        </AbstractNav>
                                      </ForwardRef(ContextTransform(AbstractNav))>
                                    </Nav>
                                  </ForwardRef(Bootstrap(Nav))>
                                </Uncontrolled(Component)>
                              </ForwardRef>
                            </ForwardRef(ContextTransform)>
                          </div>
                        </Transition>
                      </Collapse>
                    </NavbarCollapse>
                  </ForwardRef(Bootstrap(NavbarCollapse))>
                </div>
              </Container>
            </ForwardRef(Bootstrap(Container))>
          </nav>
        </Navbar>
      </Uncontrolled(Navbar)>
    </ForwardRef>
  </ForwardRef(Bootstrap(undefined))>
</AppHeader>
`;

exports[`AppHeader renders correctly 2`] = `
<AppHeader
  prop="value"
>
  <ForwardRef(Bootstrap(undefined))
    bg="primary"
    sticky="top"
  >
    <ForwardRef
      bg="primary"
      bsPrefix="navbar"
      innerRef={null}
      sticky="top"
    >
      <Uncontrolled(Navbar)
        bg="primary"
        bsPrefix="navbar"
        innerRef={null}
        sticky="top"
      >
        <Navbar
          as="nav"
          bg="primary"
          bsPrefix="navbar"
          collapseOnSelect={false}
          expand={true}
          onToggle={[Function]}
          sticky="top"
          variant="light"
        >
          <nav
            className="navbar navbar-expand navbar-light bg-primary sticky-top"
          >
            <ForwardRef(Bootstrap(Container))
              className="header"
            >
              <Container
                as="div"
                bsPrefix="container"
                className="header"
                fluid={false}
              >
                <div
                  className="header container"
                >
                  <ForwardRef(Bootstrap(NavbarBrand))>
                    <NavbarBrand
                      bsPrefix="navbar-brand"
                    >
                      <span
                        className="navbar-brand"
                      >
                        Git Cheatsheet
                      </span>
                    </NavbarBrand>
                  </ForwardRef(Bootstrap(NavbarBrand))>
                  <ForwardRef(Bootstrap(NavbarCollapse))>
                    <NavbarCollapse
                      bsPrefix="navbar-collapse"
                    >
                      <Collapse
                        appear={false}
                        dimension="height"
                        getDimensionValue={[Function]}
                        in={false}
                        mountOnEnter={false}
                        timeout={300}
                        unmountOnExit={false}
                      >
                        <Transition
                          addEndListener={[Function]}
                          appear={false}
                          aria-expanded={null}
                          enter={true}
                          exit={true}
                          in={false}
                          mountOnEnter={false}
                          onEnter={[Function]}
                          onEntered={[Function]}
                          onEntering={[Function]}
                          onExit={[Function]}
                          onExited={[Function]}
                          onExiting={[Function]}
                          timeout={300}
                          unmountOnExit={false}
                        >
                          <div
                            aria-expanded={null}
                            className="navbar-collapse collapse"
                          >
                            <ForwardRef(ContextTransform)
                              className="mr-auto"
                            >
                              <ForwardRef
                                className="mr-auto"
                                navbar={true}
                                navbarBsPrefix="navbar"
                              >
                                <Uncontrolled(Component)
                                  className="mr-auto"
                                  innerRef={null}
                                  navbar={true}
                                  navbarBsPrefix="navbar"
                                >
                                  <ForwardRef(Bootstrap(Nav))
                                    className="mr-auto"
                                    navbar={true}
                                    navbarBsPrefix="navbar"
                                    onSelect={[Function]}
                                  >
                                    <Nav
                                      as="div"
                                      bsPrefix="nav"
                                      className="mr-auto"
                                      fill={false}
                                      justify={false}
                                      navbar={true}
                                      navbarBsPrefix="navbar"
                                      onSelect={[Function]}
                                    >
                                      <ForwardRef(ContextTransform(AbstractNav))
                                        as="div"
                                        className="mr-auto navbar-nav"
                                        onSelect={[Function]}
                                      >
                                        <AbstractNav
                                          as="div"
                                          className="mr-auto navbar-nav"
                                          onSelect={[Function]}
                                          parentOnSelect={[Function]}
                                        >
                                          <div
                                            className="mr-auto navbar-nav"
                                            onKeyDown={[Function]}
                                          >
                                            <ForwardRef(Bootstrap(NavLink))>
                                              <NavLink
                                                as={[Function]}
                                                bsPrefix="nav-link"
                                                disabled={false}
                                                innerRef={null}
                                              >
                                                <AbstractNavItem
                                                  as={[Function]}
                                                  className="nav-link"
                                                  disabled={false}
                                                >
                                                  <SafeAnchor
                                                    as="a"
                                                    className="nav-link"
                                                    data-rb-event-key={null}
                                                    disabled={false}
                                                    onClick={[Function]}
                                                  >
                                                    <a
                                                      className="nav-link"
                                                      data-rb-event-key={null}
                                                      href="#"
                                                      onClick={[Function]}
                                                      onKeyDown={[Function]}
                                                      role="button"
                                                    >
                                                      About
                                                    </a>
                                                  </SafeAnchor>
                                                </AbstractNavItem>
                                              </NavLink>
                                            </ForwardRef(Bootstrap(NavLink))>
                                          </div>
                                        </AbstractNav>
                                      </ForwardRef(ContextTransform(AbstractNav))>
                                    </Nav>
                                  </ForwardRef(Bootstrap(Nav))>
                                </Uncontrolled(Component)>
                              </ForwardRef>
                            </ForwardRef(ContextTransform)>
                            <ForwardRef(ContextTransform)>
                              <ForwardRef
                                navbar={true}
                                navbarBsPrefix="navbar"
                              >
                                <Uncontrolled(Component)
                                  innerRef={null}
                                  navbar={true}
                                  navbarBsPrefix="navbar"
                                >
                                  <ForwardRef(Bootstrap(Nav))
                                    navbar={true}
                                    navbarBsPrefix="navbar"
                                    onSelect={[Function]}
                                  >
                                    <Nav
                                      as="div"
                                      bsPrefix="nav"
                                      fill={false}
                                      justify={false}
                                      navbar={true}
                                      navbarBsPrefix="navbar"
                                      onSelect={[Function]}
                                    >
                                      <ForwardRef(ContextTransform(AbstractNav))
                                        as="div"
                                        className="navbar-nav"
                                        onSelect={[Function]}
                                      >
                                        <AbstractNav
                                          as="div"
                                          className="navbar-nav"
                                          onSelect={[Function]}
                                          parentOnSelect={[Function]}
                                        >
                                          <div
                                            className="navbar-nav"
                                            onKeyDown={[Function]}
                                          >
                                            <ForwardRef(Bootstrap(NavLink))
                                              href="https://github.com/kmingulov/git-cheatsheet"
                                              target="_blank"
                                            >
                                              <NavLink
                                                as={[Function]}
                                                bsPrefix="nav-link"
                                                disabled={false}
                                                href="https://github.com/kmingulov/git-cheatsheet"
                                                innerRef={null}
                                                target="_blank"
                                              >
                                                <AbstractNavItem
                                                  as={[Function]}
                                                  className="nav-link"
                                                  disabled={false}
                                                  href="https://github.com/kmingulov/git-cheatsheet"
                                                  target="_blank"
                                                >
                                                  <SafeAnchor
                                                    as="a"
                                                    className="nav-link"
                                                    data-rb-event-key="https://github.com/kmingulov/git-cheatsheet"
                                                    disabled={false}
                                                    href="https://github.com/kmingulov/git-cheatsheet"
                                                    onClick={[Function]}
                                                    target="_blank"
                                                  >
                                                    <a
                                                      className="nav-link"
                                                      data-rb-event-key="https://github.com/kmingulov/git-cheatsheet"
                                                      href="https://github.com/kmingulov/git-cheatsheet"
                                                      onClick={[Function]}
                                                      onKeyDown={[Function]}
                                                      target="_blank"
                                                    >
                                                      <GoMarkGithub
                                                        size={24}
                                                      >
                                                        <div />
                                                      </GoMarkGithub>
                                                    </a>
                                                  </SafeAnchor>
                                                </AbstractNavItem>
                                              </NavLink>
                                            </ForwardRef(Bootstrap(NavLink))>
                                          </div>
                                        </AbstractNav>
                                      </ForwardRef(ContextTransform(AbstractNav))>
                                    </Nav>
                                  </ForwardRef(Bootstrap(Nav))>
                                </Uncontrolled(Component)>
                              </ForwardRef>
                            </ForwardRef(ContextTransform)>
                          </div>
                        </Transition>
                      </Collapse>
                    </NavbarCollapse>
                  </ForwardRef(Bootstrap(NavbarCollapse))>
                </div>
              </Container>
            </ForwardRef(Bootstrap(Container))>
          </nav>
        </Navbar>
      </Uncontrolled(Navbar)>
    </ForwardRef>
  </ForwardRef(Bootstrap(undefined))>
</AppHeader>
`;
